providers/implementations/kdfs/pbkdf2.c

```diff
@@ -36,6 +36,29 @@
 #define KDF_PBKDF2_MAX_KEY_LEN_DIGEST_RATIO 0xFFFFFFFF
 #define KDF_PBKDF2_MIN_ITERATIONS 1000
 #define KDF_PBKDF2_MIN_SALT_LEN   (128 / 8)
+/* The Implementation Guidance for FIPS 140-3 says in section D.N
+ * "Password-Based Key Derivation for Storage Applications" that "the vendor
+ * shall document in the module’s Security Policy the length of
+ * a password/passphrase used in key derivation and establish an upper bound
+ * for the probability of having this parameter guessed at random. This
+ * probability shall take into account not only the length of the
+ * password/passphrase, but also the difficulty of guessing it. The decision on
+ * the minimum length of a password used for key derivation is the vendor’s,
+ * but the vendor shall at a minimum informally justify the decision."
+ *
+ * ACVP may assume 8, most FIPS modules choose 8, BC-FJA chose 14.
+ *
+ * Allow setting this for default provider too, in case consistency is
+ * desired for FIPS and Default providers. Because password being
+ * accepted on one system, but not the other, is very confusing.
+ */
+#ifndef KDF_PBKDF2_MIN_PASSWORD_LEN
```

Comment:
To avoid bytes/bits confusion, would it be helpful to call this `KDF_PBKDF2_MIN_PASSWORD_BYTES`? and have the parameter to `pbkdf2_lower_bound_check_passed()` be `passbytes` rather than `passlen`?

Reply:
throughout codebase _LEN is used for byte-length; and _LEN_BITS is used for bit-lengths.
also passlen is used throughout the codebase, but not passbytes.
I do agree that in the UTF-8 world passbytes != passlen, but that would need codebase wide decision and change on how to count UTF-8 password lengths.